File: go-src/internal/slack/mrkdwn_test.go
package slack

import "testing"

func TestFixSlackMrkdwn(t *testing.T) {
	tests := []struct {
		name string
		in   string
		want string
	}{
		{
			name: "FullwidthColonAfterBold",
			in:   "*住所*：東京",
			want: "*住所*: 東京",
		},
		{
			name: "FullwidthParenAfterBold",
			in:   "*金融ch*（17件）",
			want: "*金融ch* （17件）",
		},
		{
			name: "FullwidthBracketAfterBold",
			in:   "*重要*「注意」",
			want: "*重要* 「注意」",
		},
		{
			name: "FullwidthCommaAfterBold",
			in:   "*項目*、次",
			want: "*項目* 、次",
		},
		{
			name: "AsciiAfterBold_NoChange",
			in:   "*bold* text",
			want: "*bold* text",
		},
		{
			name: "HalfwidthColonAfterBold_NoChange",
			in:   "*label*: value",
			want: "*label*: value",
		},
		{
			name: "MultipleFixesInText",
			in:   "*住所*：東京\n*金額*（100万円）",
			want: "*住所*: 東京\n*金額* （100万円）",
		},
		{
			name: "NonBoldAsterisk_NoChange",
			in:   "5 * 3 = 15：答え",
			want: "5 * 3 = 15：答え",
		},
		{
			name: "DoubleAsteriskToSingle",
			in:   "**太字**テスト",
			want: "*太字* テスト",
		},
		{
			name: "EmptyString",
			in:   "",
			want: "",
		},
		{
			name: "DoubleAsteriskWithFullwidthColon",
			in:   "**見出し**：内容",
			want: "*見出し*: 内容",
		},
		{
			name: "EmojiAfterBold",
			in:   "*結果*🔴失敗",
			want: "*結果* 🔴失敗",
		},
		{
			name: "MultipleBoldsOnSameLine",
			in:   "*A*（1）と*B*（2）",
			want: "*A* （1）と*B* （2）",
		},
		{
			name: "OnlyASCII_NoChange",
			in:   "Hello *world* test",
			want: "Hello *world* test",
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := FixSlackMrkdwn(tt.in)
			if got != tt.want {
				t.Errorf("FixSlackMrkdwn(%q)\n got  = %q\n want = %q", tt.in, got, tt.want)
			}
		})
	}
}

File: src/lib/__tests__/mrkdwn.test.ts
import { describe, it, expect } from "vitest";
import { fixSlackMrkdwn } from "../mrkdwn.js";

describe("fixSlackMrkdwn", () => {
  // Go 版テストケース完全移植 (14件)
  const cases: { name: string; input: string; expected: string }[] = [
    {
      name: "FullwidthColonAfterBold",
      input: "*住所*：東京",
      expected: "*住所*: 東京",
    },
    {
      name: "FullwidthParenAfterBold",
      input: "*金融ch*（17件）",
      expected: "*金融ch* （17件）",
    },
    {
      name: "FullwidthBracketAfterBold",
      input: "*重要*「注意」",
      expected: "*重要* 「注意」",
    },
    {
      name: "FullwidthCommaAfterBold",
      input: "*項目*、次",
      expected: "*項目* 、次",
    },
    {
      name: "AsciiAfterBold_NoChange",
      input: "*bold* text",
      expected: "*bold* text",
    },
    {
      name: "HalfwidthColonAfterBold_NoChange",
      input: "*label*: value",
      expected: "*label*: value",
    },
    {
      name: "MultipleFixesInText",
      input: "*住所*：東京\n*金額*（100万円）",
      expected: "*住所*: 東京\n*金額* （100万円）",
    },
    {
      name: "NonBoldAsterisk_NoChange",
      input: "5 * 3 = 15：答え",
      expected: "5 * 3 = 15：答え",
    },
    {
      name: "DoubleAsteriskToSingle",
      input: "**太字**テスト",
      expected: "*太字* テスト",
    },
    {
      name: "EmptyString",
      input: "",
      expected: "",
    },
    {
      name: "DoubleAsteriskWithFullwidthColon",
      input: "**見出し**：内容",
      expected: "*見出し*: 内容",
    },
    {
      name: "EmojiAfterBold",
      input: "*結果*🔴失敗",
      expected: "*結果* 🔴失敗",
    },
    {
      name: "MultipleBoldsOnSameLine",
      input: "*A*（1）と*B*（2）",
      expected: "*A* （1）と*B* （2）",
    },
    {
      name: "OnlyASCII_NoChange",
      input: "Hello *world* test",
      expected: "Hello *world* test",
    },
  ];

  for (const { name, input, expected } of cases) {
    it(name, () => {
      expect(fixSlackMrkdwn(input)).toBe(expected);
    });
  }
});
